### DISCUSSION TITLE
Is it possible to build new scenery library with some files exclusion? Or add new scenery faster?

### DISCUSSION THREAD

**Discussion by BattleOverlord** [albar965/littlenavmap#1191] [Q&A]
First of all I'm not sure if this is Q&A or sharing an idea. I chose up Q&A and hopefully you will understand this question. I will try to make it as brief and clear.

I use a lot of freeware and payware airports,. I use addon linker to de/activate addons quite often. To cut loading times I usually have 20-30 addons loaded (5-10 scenery/airports).

Imagine this scenario:
1) I have custom airports I plan flight between. Lets say Frankfurt to Rome.
2) I use addon linker to activate both airport folders
3) I run the LNAVMAP - LNAVAMAP shows default airports (both of them)
4) I run the "Load flight simulator scenery library into littlenavmap database" in LNAVMAP
5) Now LNAVMAP shows these two recently added airports correctly
_______________________________________________________________________________________________
6) I do the flight. I turn off the game. Now I want to do flight for example from Addis Ababa to Cape Town
7) I use addon linker to activate both airport folders and DEACTIVATE Franfkurt and Rome airports.
8) 3) I run the LNAVMAP - LNAVMAP shows 2 airports like the addon would be still active (Franfkurt and Rome airports). and it shows Addis Ababa and Cape town airports as default.
9) I run the "Load flight simulator scenery library into littlenavmap database" in LNAVMAP
10) Now it obviously changes and LNAVMAP shows 2 airports correctly - Addis Ababa and Cape town airports AND airports Franfkurt and Rome  as default.

Is there a way how to not overwrite the "not currently active once already added custom airports"? Or how to exclude it from "refreshing" the scenery because of the Load flight simulator scenery library into littlenavmap database process?

In other words I would like to build new scenery file everytime I add new airportin for the first time,  but when I deactivate it I would like to have it still visible in LNAVMAP because when I activate these airports again (for example forementioned Franfkurt and Rome) I would not have to use the Load flight simulator scenery library into littlenavmap database function again. I would like to add new custom airports to that database and once they are in there they would stay there even if I build the database again untill I delete/refresh it manually or cancel some exclusion/exemption.

Hope you understand this idea/question/request. If you would need more explanation in this matter I'll do my best.

Thank you very much.

**Comment by albar965** [comment]
Not sure if I understand but why not create an add-on linker profile containing all airports. Then let LNM load the scenery library. This step can be done (not required) each time a new airport is added.

For flying you can enable or disable whatever airports you have. This won't affect LNM since it keeps all information stored in its own database.

**Reply by BattleOverlord** [reply]
> Not sure if I understand but why not create an add-on linker profile containing all airports. Then let LNM load the scenery library. This step can be done (not required) each time a new airport is added.
> 
> For flying you can enable or disable whatever airports you have. This won't affect LNM since it keeps all information stored in its own database.

This idea came to my mind as well. Activate every airport, build the database in LNAVMAP and leave it. Of course whenever I add new airport which is frankly quite often I would have to repeat this step many many times. It would be cool if I could tell the LNAVMAP ... hey I have new airport and its located here load it in, dont care about the rest of the airports. It would be loaded/added into the database within seconds. Now it takes about 5 minutes to rebuild the database.

**Reply by albar965** [reply]
Reloading of single airports is not doable due to too many cross references between airports, navaids and procedures.

Reloading is not really required. At worst aprons and parking spots don't match but you can still use the airport forplanning a flight. 
BTW: You can also plan a flight while reloading the scenery library in the background.

**Reply by BattleOverlord** [reply]
Ok. I understand. Well to me LNAVMAP is more than a planning tool. It is versatile powerful transformer :) I agree it is not required to reload it each time, but it is very handy when the ground, taxiways and aprons are also updated to the addon airport version, because I use also LNAVMAP for taxiing because I have no navigraph subscription (atc assigns me new gate/ramp which corresponds with the addon airport). Also I'm not sure about this one, but some custom airports also change the navdata heavily (sids, stars, approaches), if for example Asobo did not add ILS from both sides and in real life the runway has ils from both runways it also shows in LNAVMAP when you update the scenery database.
I will probably activate all airports once a week or month and rebuild my LNAVMAP database. Thanks :)
